environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java

```diff
@@ -135,10 +140,28 @@ private void editDescriptionIfChanged(Environment environment, EnvironmentEditDt
         }
     }
 
-    private void editAuthenticationIfChanged(EnvironmentEditDto editDto, Environment environment) {
+    @VisibleForTesting
+    void editAuthenticationIfChanged(EnvironmentEditDto editDto, Environment environment) {
         AuthenticationDto authenticationDto = editDto.getAuthentication();
         if (authenticationDto != null) {
+            EnvironmentValidatorService validatorService = environmentService.getValidatorService();
+            ValidationResult validationResult = validatorService.validateAuthenticationModification(editDto, environment);
+            if (validationResult.hasError()) {
+                throw new BadRequestException(validationResult.getFormattedErrors());
+            }
+            String oldSshKeyIdForDeletion = null;
+            String oldSshKeyId = environment.getAuthentication().getPublicKeyId();
+            if (environment.getAuthentication().isManagedKey()) {
```

Comment:
how about if we ccreate a method in environement and then calling like environment.isEnvironmentMAnagedKEy() ?

Reply:
the method name does not describe well the functionality.